hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +301,461 @@ The following table lists the main parameters used throughout the OTP controller
+The key used for scrambling is a global netlist constant chosen by the silicon creator, and all secret partitions are encrypted using the same netlist constant.
+Note that the amount of data that is being scrambled is small (160 Byte = 20 x 64bit blocks) and the scrambled data remains constant.
+Hence, no additional masking or diversification scheme is applied since only a very limited amount of information can be gathered by observing the scrambling operation via side-channels.
+
+**TBD: should each secret partition have its own scrambling key constant?**
+
+#### Digest Calculation Mode
+
+The integrity digests used in the [partition checks]({{< relref "#partition-checks" >}}) are computed using a custom [Merkle-Damgard](https://en.wikipedia.org/wiki/Merkle%E2%80%93Damg%C3%A5rd_construction) construction as illustrated in subfigure b).
+At the beginning of the digest calculation the 64bit state is initialized with an initialization vector (IV).
+Then, the data to be digested is split into 128bit chunks, each of which is used as a 128bit key input for updating the 64bit state with seven PRESENT encryption rounds.
+Chunks that are not aligned with 128bit are padded with zero, and the finalization operation consists of another seven-round encryption pass with a finalization constant.
+Note that both the IV as well as the finalization constant are global netlist constants chosen by the silicon creator.
+
+#### Ephemeral Key Derivation Mode for SRAM
+
+The key derivation modes for ephemeral SRAM scrambling keys leverages the digest mechanism as illustrated in subfigure c).
+In particular, the ephemeral key is derived by repeatedly reducing a semi-random 512bit block of data into a 64bit digest.
+The 512bit block is composed of the 128bit SRAM_DATA_KEY stored in OTP, as well as 384bit of fresh entropy fetched from the CSRNG.
+Depending on the ephemeral key width (128bit for SRAM, or 512bit for OTBN), this process has to be repeated 2 or 8 times.
+The IV and finalization constants for this key derivation are different from the ones used for the digest mode.
+
+#### Static Key Derivation Mode for FLASH
+
+The static key derivation mode for FLASH scrambling keys leverages the PRESENT encryption mechanism, as shown in subfigure d).
+E.g., in order to produce the lower 64bit of the FLASH data scrambling key, an IV is encrypted using the lower 128bit of the 256bit FLASH_DATA_KEY stored in OTP.
+Since both the data and address scrambling keys for FLASH are 128bit in size, that makes for a total of four 31-round encryption passes that are needed to produce these static keys.
+Note that each of these passes uses a separate, 64bit IV netlist constant chosen by the silicon creator.
+
+**TODO: need to align the scrambling key sizes in OTP for this to be correct**
+
+### Primitive Wrapper and FPGA Emulation
+
+![OTP Wrapper Block Diagram](otp_wrapper_blockdiag.svg)
```

Comment:
image source here for reference:
https://docs.google.com/drawings/d/1fNXqDXGOR7iLRvLpSWunaQznHHbNEMubGmvAU7bwxPo